clloader/scenarios/base.py

```diff
@@ -93,22 +93,40 @@ def __getitem__(self, task_index):
             data_type=self.cl_dataset.data_type
         )
 
-    def _select_data_by_task(self, ind_task: int):
+    def _select_data_by_task(self, task_index: Union[int, slice]):
         """Selects a subset of the whole data for a given task.
 
         :param ind_task: task index
         :return: A tuple of numpy array, the first item being the data and the
                  second the associated targets.
         """
-        x_, y_, t_ = self.dataset  # type: ignore
-
-        indexes = np.where(t_ == ind_task)[0]
-        selected_x = x_[indexes]
-        selected_y = y_[indexes]
+        x, y, t = self.dataset  # type: ignore
+
+        if isinstance(task_index, slice):
+            start = task_index.start or 0
+            stop = task_index.stop
+            step = task_index.step or 1
+            task_indexes = list(range(start, stop, step))
+            task_indexes = [
+                t if t >= 0 else _handle_negative_indexes(t, len(self)) for t in task_indexes
+            ]
+            indexes = np.where(np.isin(t, task_indexes))[0]
+        else:
+            if task_index < 0:
+                task_index = _handle_negative_indexes(task_index, len(self))
+            indexes = np.where(t == task_index)[0]
+        selected_x = x[indexes]
+        selected_y = y[indexes]
```

Comment:
I think this is not compatible with the transformed incremental spirit.
In transform incremental two tasks are different because they have a differeent transformation.
Therefore, I don't think we can mix two tasks into one easily.
I thought I understood that we wanted slice to iterate on specific tasks, not to merge into one task several task.
Or maybe I miss something in your modif ...

Reply:
Well slicing means using several indexes at once.

I don't a big problem for transformed to "slice" several tasks together, but if you think that's a problem, you can raise an error when using a slice for only this scenario.

It make sense in other scenario. For exemple it's now really clean to evaluate on all seen tasks by simply doing `clloader[:task_id]` instead of looping.

Reply:
To raise an error in transformed incremental, what I would do is:

Override the getitem method, if the input is a slice, raise an error, else call the parent getitem method.

Reply:
And actually, it would also be a good idea that the dataset provide in addition of `x` and `y`, `t`.

It can be ignored when doing `clloader[task_id]`, as all `t` would be `task_id`. But when slicing it may be important. And in this case, I don't really see a problem for transformed.

And if that really a big deal, people using transformed simply shoudln't use slicing.

Reply:
ok then :)